tools/nl/loadtest/load.py

```diff
@@ -0,0 +1,157 @@
+  try:
+    response = session.get(_URL + query, timeout=None)
+    # response = session.get(_URL, timeout=None)
+    if response.status_code == 200:
+      debug_timing = response.json().get('debug', {}).get('timing', {})
+      return {
+          _RESULT_KEY_CODE: str(response.status_code),
+          _RESULT_KEY_NL_TIME: debug_timing.get('getNlResult'),
+          _RESULT_KEY_TOTAL_TIME: debug_timing.get('total'),
+          _RESULT_KEY_TILE_TIME: debug_timing.get('getTileResults')
+      }
+    return {
+        _RESULT_KEY_CODE: str(response.status_code),
+        _RESULT_KEY_ERROR_CONTENT: str(response.content)
+    }
+  except Exception as e:
+    return {_RESULT_KEY_CODE: 'Exception', _RESULT_KEY_ERROR_CONTENT: str(e)}
+
+
+# Get a list of queries
+def _get_query_list():
+  query_list = []
+  with open(_QUERY_FILE) as inf:
+    for row in csv.reader(inf):
+      if not row:
+        continue
+      query = row[0].strip()
+      if not query or query.startswith('#') or query.startswith('//'):
+        continue
+      assert ';' not in query, 'Multiple query not yet supported'
+      query_list.append(query)
+  return query_list
+
+
+# Take the list of results from running the queries and get the result output
+# object.
+def _get_result_output(run_results):
+  response_codes = {}
+  times_seen = {}
+  result = {
+      _RESULT_KEY_NL_TIME: {
+          'avg': 0,
+          'max': 0,
+          'min': 0
+      },
+      _RESULT_KEY_TILE_TIME: {
+          'avg': 0,
+          'max': 0,
+          'min': 0
+      },
+      _RESULT_KEY_TOTAL_TIME: {
+          'avg': 0,
+          'max': 0,
+          'min': 0
+      },
+  }
+  error_content = {}
+  for r in run_results:
+    code = r.get('code')
+    occ = response_codes.get(code, 0) + 1
+    response_codes[code] = occ
+    if r.get('errorContent'):
+      if not code in error_content:
+        error_content[code] = {}
+      content = r.get('errorContent')
+      content_occurence = error_content[code].get(content, 0) + 1
+      error_content[code][content] = content_occurence
+    for k in [
+        _RESULT_KEY_NL_TIME, _RESULT_KEY_TILE_TIME, _RESULT_KEY_TOTAL_TIME
+    ]:
+      if not r.get(k):
+        continue
+      num_seen = times_seen.get(k, 0) + 1
+      times_seen[k] = num_seen
+      time = float(r.get(k))
+      result[k]['max'] = max(time, result[k]['max'])
+      if result[k]['min'] == 0:
+        result[k]['min'] = time
+      else:
+        result[k]['min'] = min(time, result[k]['min'])
+      result[k]['avg'] = result[k]['avg'] * (num_seen -
+                                             1) / num_seen + time / num_seen
+  result['codes'] = response_codes
+  result['errors'] = error_content
+  return result
+
+
+# Runs a list of queries against _URL.
+def run_load(total_requests, parallel_requests):
+  query_list = _get_query_list()
+  query_list_to_run = []
+  # Using the query_list, generate a list of queries that is total requests long
```

Comment:
super nit: some vertical space between code blocks for readability?

Reply:
done